testutil/base.go

```diff
@@ -60,3 +62,31 @@ func (s *BaseTest) TearDownTest(c *check.C) {
 func (s *BaseTest) AddCleanup(f func()) {
 	s.cleanupHandlers = append(s.cleanupHandlers, f)
 }
+
+// Mocks any mutable object with a compliant substitute, and insures it will be restored after the test
```

Comment:
nitpick: I'd call it a variable instead of a mutable object
```suggestion
// Mocks variable with a compliant substitute, and insures it will be restored after the test
```

Reply:
@MiguelPires, I think specifying mentioning "**mutable**" is important, you can't just patch anything, but things that are allowed to be patched.

Reply:
"variable" is a synonym of "mutable". I also think the reader understands that you can only change things that can be changed. But feel free to keep it as you prefer

Reply:
@MiguelPires, from my experience with functional programming and the literature on this subject, I saw that immutable variable is a thing. I understand that it sounds weird. But there is such things as _mutable variable_ and _immutable variable_. Moreover, in some languages (dynamic languages as far as I know) there is a mocking method called monkey-patching, which actually allows substituting the immutable staff as well. There is even an attempt to implement [monkey-patching for Golang](https://github.com/bouk/monkey), but it considered a bit problematic and coupled with specific CPU architecture only. It won't work for us.
I have a slight preference to keep mutable objects or mutable variables, then just variables to avoid confusion to those who might anticipate monkey-patching abilities in these methods.